config/manager/manager.yaml

```diff
@@ -35,6 +35,10 @@ spec:
         name: manager
         securityContext:
           allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+              - ALL
+          readOnlyRootFilesystem: true
```

Comment:
please add reference where it comes from

Reply:
For some context, the controller have some security findings need to be solved, one of it is to prevent privilege escalation, by suggestions in that security findings ticket and what other project do, for example:

https://github.com/kubernetes-sigs/external-dns/blob/master/charts/external-dns/values.yaml#L54-L60
https://github.com/projectcontour/contour/blob/main/examples/render/contour-deployment.yaml#L8398-L8401


We could add following option to prevent escalation and make sure not using the root user to run the controller  

prevent root user already defined by https://github.com/aws/aws-application-networking-k8s/blob/main/Dockerfile#L25-L28 (not UID and GID 0) and https://github.com/aws/aws-application-networking-k8s/blob/634bad29f0473bc100d428b6e1cbb31a7284203b/examples/deploy-v0.0.18.yaml#L7850-L7851 
and we could add this in the helm template (pod level securityContext):
```
     securityContext:
        runAsNonRoot: true
```

by https://github.com/kubernetes-sigs/external-dns/blob/master/charts/external-dns/values.yaml#L54-L60
and https://github.com/projectcontour/contour/blob/main/examples/render/contour-deployment.yaml#L8398-L8401
We could add these setting, to make the controller process don't have not necessary permissions (container level securityContext)
```
        securityContext:
          allowPrivilegeEscalation: false
          capabilities:
            drop:
              - ALL
          readOnlyRootFilesystem: true
```

@solmonk  should have more context on it, could you take a look this PR?

Reply:
* I don't think we need an explanation inside the code, it is pretty common over the k8s world, and we are responsible for setting right capabilities of the container.
* Have you double checked that the updated capabilities are sufficient? (I think so though)

Reply:
I think capabilities change is ok. I deployed the controller by `kubectl apply  -f deploy.yaml` and did some manually tests, controller logs look good, nothing abnormal.

Reply:
Thank you, I didnt mean to add reference to code, but to PR, sorry. Few questions
1. How did you test privilege escalation?
2. What "capabilities drop all" means?

Reply:
We already used the nonroot base image, so the controller process container is already not able to get root permission https://github.com/aws/aws-application-networking-k8s/blob/f5f17fd6da69116c2d9eadca164262355d65eb14/Dockerfile#L25 
Also the `allowPrivilegeEscalation: false` and `runAsNonRoot: true` flags double confine that
(the [distroless](https://github.com/GoogleContainerTools/distroless) base image is so minimal I am quite hard to test it)


Drop ALL means the controller process don't have capabilities that I showed in the PR description.
_From https://kubernetes.io/docs/tasks/configure-pod-container/security-context,
[Linux Capabilities](https://linux-audit.com/linux-capabilities-hardening-linux-binaries-by-removing-setuid/): Give a process some privileges, but not all the privileges of the root user._

For example, it will drop the CAP_NET_BIND_SERVICE (capabilityof binding ports below 1024):
```
~ $ nc -l -p 80
nc: bind: Permission denied
```
for the process has this capability:
```
nc -l -p 80

netstat -tuln
Proto Recv-Q Send-Q Local Address           Foreign Address         State       
tcp        0      0 :::80                   :::*                    LISTEN
```
And for the flag `readOnlyRootFilesystem: true` control this:
```
touch /tmp/testfile
touch: /tmp/testfile: Read-only file system
```


I used the debug version of distroless to test them of these flags
```
apiVersion: v1
kind: Pod
metadata:
  name: my-test-pod-non-privilege
spec:
  securityContext:
    runAsNonRoot: true
  containers:
    - name: my-test-pod-non-privilege
      image: gcr.io/distroless/static-debian12:debug-nonroot
      command: ["sh", "-c", "while true; do sleep 1; done"]
      env:
        - name: PodName
          value: "my-test-pod-non-privilege handler pod"
      securityContext:
        allowPrivilegeEscalation: false
        capabilities:
          drop:
            - ALL
        readOnlyRootFilesystem: true
```

Reply:
thank you! looks good to me